cells/src/commonMain/kotlin/com/wire/kalium/cells/domain/usecase/DownloadCellFileUseCase.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.cells.domain.usecase
+
+import com.wire.kalium.cells.domain.CellsRepository
+import com.wire.kalium.common.functional.map
+import com.wire.kalium.common.functional.onFailure
+import com.wire.kalium.common.functional.onSuccess
+import com.wire.kalium.logic.data.asset.AssetTransferStatus
+import com.wire.kalium.util.KaliumDispatcher
+import com.wire.kalium.util.KaliumDispatcherImpl
+import kotlinx.coroutines.withContext
+import okio.Path
+
+public class DownloadCellFileUseCase internal constructor(
+    private val cellsRepository: CellsRepository,
```

Comment:
We have some tests on `logic` that check for things like "data layer can't depend on use cases" and "all use cases must be documented", etc.

I just realised that with modularisation, this can be left without check. I don't think it's a rule that is _always_ necessary. But it's better to be 10% annoying than 90% missing in documentation. This usecase seems pretty self-explanatory. 


In general, with this comment, I just want to raise awareness about the tests we had on `logic` and that are no longer being checked in feature modules.

Something that we - as a team - should resolve at some point.